### DISCUSSION TITLE
How do you log when using TaskFlow and a Virtualenv?

### DISCUSSION THREAD

**Discussion by ADraginda** [apache/airflow#34948] [Q&A]
I have a dag that has the task:

```python
@task.virtualenv(requirements=[...], system_site_packages=False)
def foo():
    import logging

    logging.info('base logger')

    logger = logging.getLogger("airflow.task")
    logger.info('airflow.task logger')

    from airflow.utils.log.logging_mixin import LoggingMixin
    logger_mixin = LoggingMixin().log
    logger_mixin.info('mixin logger')
    
    print('just a print')
```

Output

```
[2023-10-14, 23:13:15 UTC] {{process_utils.py:189}} INFO - just a print
```

So obviously I'm logging incorrectly since the one message was the print statement. The [TaskFlow](https://airflow.apache.org/docs/apache-airflow/stable/core-concepts/taskflow.html) instructions say to use `logger = logging.getLogger("airflow.task")` but this recommendation doesn't seem to work in virtualenvs?

Any suggestions on what is the right way to log within a TaskFlow virtualenv? (Strangely enough, I do not have this issue with PythonVirtuanevOperator - just the taskflow flavor).

Thanks!

**Comment by ADraginda** [comment]
Turned out to be a logging config issue on our side, logging via `airflow.task` was actually working just fine. Closing as resolved since my sample code actually works just fine

**Comment by Zifeng-Ma** [comment]
Hello, I have similar issue nowadays.
I am also using `@task.virtualenv`
```
@task.virtualenv(
    python_version="3.9",
    requirements="requirements_example.txt",  # Must exist in DAG folder; list all packages needed in virtualenv
    system_site_packages=True,
    venv_cache_path="venvs/example",          # Must exist; caches virtualenv to speed up subsequent runs
    inherit_env=True,
)
def task_two(logical_date=None, params=None):
    import logging
    logger = logging.getLogger("airflow.task")
    logger.info('airflow.task logger')
```

But it doesnt work, can you please tell me how did you change the config to make it work? Thank you!
